urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/price/Price.java

```diff
@@ -0,0 +1,290 @@
+/**
+ * Price.java
+ */
+package price;
+
+import price.utils.InvalidPriceOperation;
+
+/**
+ * This class governs the construction and operation of a Price object.<br> This
+ * object is immutable and cannot be changed after creation.
+ *
+ * @author Kevin Patrick Westropp
+ * @version StockExVirtuoso.Version.1.0
+ */
+public final class Price implements Comparable<Price> {
+
+    /**
+     * Value of price.
+     */
+    private final long VALUE;
+    /**
+     * Boolean Market Price: Yes or No?.
+     */
+    private final boolean MKT;
+
+    /**
+     * Package-visible Construction Method for a new Limit Price object.
+     *
+     * @param value Price value passed into setValue
+     */
+    Price(long value) {
+        this.VALUE = value;
+        this.MKT = false;
+    }
+
+    /**
+     * Package-visible Construction Method for a new Market Price object.
+     *
+     * @see #setValue(long)
+     * @see #setMarket(boolean)
+     */
+    Price() {
+        this.VALUE = 0;
+        this.MKT = true;
+    }
+
+    /**
+     * Private Get method to return value of current object.
+     *
+     * @return currentValue
+     */
+    private long getValue() {
+        return VALUE;
+    }
+
+    /**
+     * Private Get method to return true is Market Price.
+     *
+     * @return currentMKT
+     */
+    private boolean getMarketPrice() {
+        return MKT;
+    }
+
+    /**
+     * Public Add method which adds two prices and returns new price.
+     *
+     * @param p Price object
+     * @return Price object
+     * @throws InvalidPriceOperation if either of the objects are a Market Price
+     */
+    public Price add(Price p) throws InvalidPriceOperation {
+        if (isMarket() == true || p.isMarket() == true) {
+            throw new InvalidPriceOperation("InvalidPriceOperation: Cannot add a LIMIT price to a MARKET Price");
+        } else {
+            long currentValue = getValue();
+            long addValue = p.VALUE;
+            long newValue = (currentValue + addValue);
+            Price newPrice = PriceFactory.makeLimitPrice(newValue);
+            return newPrice;
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 75:81, 94:100.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.